app/code/core/Mage/GoogleAnalytics/Block/Ga.php

```diff
@@ -137,61 +137,50 @@ protected function _getEnhancedEcommerceDataForAnalytics4()
          *
          * @link https://developers.google.com/tag-platform/gtagjs/reference/events#remove_from_cart
          */
-        $removedProducts = Mage::getSingleton('core/session')->getRemovedProductsCart();
+        $removedProducts = Mage::getSingleton('core/session')->getRemovedProductsForAnalytics();
         if ($removedProducts) {
             foreach ($removedProducts as $removedProduct) {
-                $_removedProduct = Mage::getModel('catalog/product')->load($removedProduct);
                 $eventData = [];
                 $eventData['currency'] = Mage::app()->getStore()->getCurrentCurrencyCode();
-                $eventData['value'] = number_format($_removedProduct->getFinalPrice(), 2, '.', '');
+                $eventData['value'] = number_format($removedProduct['price'] * $removedProduct['qty'], 2, '.', '');
                 $eventData['items'] = [];
                 $_item = [
-                    'item_id' => $_removedProduct->getSku(),
-                    'item_name' => $_removedProduct->getName(),
-                    'price' => number_format($_removedProduct->getFinalPrice(), 2, '.', ''),
+                    'item_id' => $removedProduct['sku'],
+                    'item_name' => $removedProduct['name'],
+                    'price' => number_format($removedProduct['price'], 2, '.', ''),
+                    'quantity' => $removedProduct['qty'],
```

Comment:
add specific cast to int

`'quantity' => (int) $removedProduct['qty'],`

Reply:
done